recipes/openssl/ALL/conanfile.py

```diff
@@ -517,9 +521,28 @@ def build(self):
                     self._patch_makefile_org()
                 self._make()
 
+    @staticmethod
```

Comment:
Does all this code smell like something to fix in Conan? Something like `tools.cross_building(self.settings, native_x86=True)` or similar?

Reply:
As far as I am concerned, this is a missing feature in conan. I had to implement similar conditions several times already.

Reply:
Could you open an issue, please?

Reply:
if we agree, I am going to open an issue in conan. I had to apply same in `icu`, for example

Reply:
For the future, if you are repeating the same workaround talk with us!

Reply:
opened https://github.com/conan-io/conan/issues/5870